runtime/calamari/tests/xcm_tests.rs

```diff
@@ -0,0 +1,1791 @@
+			parachain::Origin::root(),
+			// This need to be changed starting from v0.9.16
+			// need to use something like MultiLocation { parents: 0, interior: here} instead
+			source_location.clone(),
+			asset_metadata.clone()
+		));
+		assert_ok!(AssetManager::set_units_per_second(
+			parachain::Origin::root(), 
+			a_currency_id, 
+			0u128));
+		assert_eq!(
+			Some(a_currency_id),
+			AssetManager::location_asset_id(source_location.clone())
+		);
+	});
+
+	// We don't specify units_per_second on B
+	ParaB::execute_with(|| {
+		assert_ok!(AssetManager::register_asset(
+			parachain::Origin::root(),
+			source_location.clone(),
+			asset_metadata.clone()
+		));
+		assert_eq!(
+			Some(a_currency_id),
+			AssetManager::location_asset_id(source_location.clone())
+		);
+	});
+
+	let dest = MultiLocation {
+		parents: 1,
+		interior: X2(
+			Parachain(2),
+			AccountId32 {
+				network: NetworkId::Any,
+				id: ALICE.into(),
+			},
+		),
+	};
+
+	// Transfer ParaA balance to B
+	ParaA::execute_with(|| {
+		assert_ok!(parachain::XTokens::transfer(
+			parachain::Origin::signed(ALICE.into()),
+			parachain::CurrencyId::MantaCurrency(a_currency_id),
+			amount,
+			Box::new(VersionedMultiLocation::V1(dest)),
+			dest_weight,
+		));
+		assert_eq!(
+			parachain::Balances::free_balance(&ALICE.into()),
+			INITIAL_BALANCE - amount
+		)
+	});
+
+	// Alice on B should receive nothing since we didn't specify the unit per second
+	ParaB::execute_with(|| {
+		assert_eq!(parachain::Assets::balance(a_currency_id, &ALICE.into()), 0);
+	});
+}
+
+#[test]
+fn receive_asset_with_is_sufficient_false(){
+	MockNet::reset();
+
+	let new_account = [5u8; 32];
+	let relay_asset_id = 0u32;
+	let source_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::parent()));
+	let asset_metadata = parachain::AssetRegistarMetadata {
+		name: b"Kusama".to_vec(),
+		symbol: b"KSM".to_vec(),
+		decimals: 12,
+		min_balance: 1u128,
+		evm_address: None,
+		is_frozen: false,
+		is_sufficient: false,
+	};
+	let amount = 123u128;
+
+	// register relay asset in parachain A
+	ParaA::execute_with(|| {
+		assert_ok!(parachain::AssetManager::register_asset(
+			parachain::Origin::root(),
+			source_location,
+			asset_metadata
+		));
+		// we don't charge anything during test
+		assert_ok!(parachain::AssetManager::set_units_per_second(
+			parachain::Origin::root(), 
+			relay_asset_id, 
+			0u128));
+	});
+
+	let dest: MultiLocation = AccountId32 {
+		network: Any,
+		id: new_account.into()
+	}.into();
+
+	Relay::execute_with(|| {
+		assert_ok!(RelayChainPalletXcm::reserve_transfer_assets(
+			relay_chain::Origin::signed(ALICE),
+			Box::new(X1(Parachain(1)).into().into()),
+			Box::new(VersionedMultiLocation::V1(dest.clone()).clone().into()),
+			Box::new((Here, amount).into()),
+			0,
+		));
+		assert_eq!(
+			relay_chain::Balances::free_balance(&para_account_id(1)),
+			INITIAL_BALANCE + amount
+		);
+	});
+
+	// parachain should not have received assets
```

Comment:
Why is the transfer supposed to fail here actually ?

Reply:
When receiving this assets, `ParaA` need to create a new account for `ALICE`. Due to `is_sufficient` set to `false`, parachain cannot do so. 

To the first approximation, you can think of `is_sufficient` means whether this asset can be treated as reserve asset. The logic here is, despite the receiving asset is greater than the `min_balance` of this particular asset. Since this receiving asset is not treated as reserve asset and the`ALICE` doesn't have any other kind of balances (thus the account is not `active`), this transaction should fail. 

This failure also DOES NOT trigger a refund so need to be take care of cautiously.